frontend/src/pages/functional_components/NewExperimentFC.tsx

```diff
@@ -63,6 +67,40 @@ export function NewExperimentFC(props: NewExperimentFCProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    if (pipelineId) {
+      (async () => {
+        setLatestVersion(await getLatestVersion(pipelineId));
+      })();
+    }
+  }, [pipelineId]);
+
+  // Handle the redirection work when createExperiment is succeed
+  useEffect(() => {
+    if (experimentResponse) {
+      const searchString = pipelineId
+        ? new URLParser(props).build({
+            [QUERY_PARAMS.experimentId]: experimentResponse.experiment_id || '',
+            [QUERY_PARAMS.pipelineId]: pipelineId,
+            [QUERY_PARAMS.pipelineVersionId]: latestVersion?.pipeline_version_id || '',
+            [QUERY_PARAMS.firstRunInExperiment]: '1',
+          })
+        : new URLParser(props).build({
+            [QUERY_PARAMS.experimentId]: experimentResponse.experiment_id || '',
+            [QUERY_PARAMS.firstRunInExperiment]: '1',
+          });
+      props.history.push(RoutePage.NEW_RUN + searchString);
+
+      updateSnackbar({
+        autoHideDuration: 10000,
+        message: `Successfully created new Experiment: ${experimentResponse.display_name}`,
+        open: true,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [experimentResponse]);
```

Comment:
Should we also monitor `pipelineId` in this case? It is being used by the searchString construction.